arangod/Aql/ExecutionBlockImpl.cpp

```diff
@@ -1252,12 +1252,17 @@ auto ExecutionBlockImpl<Executor>::executeProduceRows(typename Fetcher::DataRang
   if constexpr (isNewStyleExecutor<Executor>) {
     if constexpr (is_one_of_v<Executor, UnsortedGatherExecutor>) {
```

Comment:
Why is UnsortedGather special cased here?

Reply:
You added it yourself :P

UnsortedGather returns 4 outputs, instead of 3, number 4 being the dependency.